`no-undefined-types` doesn't know the `@import` types

## Expected behavior

[`no-undefined-types`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/no-undefined-types.md#readme) doesn't repport an error if the type used has been imported with [`@import` tag](https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#the-jsdoc-import-tag).

## Actual behavior

`no-undefined-types` doesn't know the `@import` types.

## Files

### `package.json`

```json
{
  "name": "testcase",
  "version": "1.0.0",
  "dependencies": {
    "eslint": "9.5.0",
    "eslint-plugin-jsdoc": "48.2.13"
  }
}
```

### `eslint.config.mjs`

```javascript
import jsdoc from "eslint-plugin-jsdoc";

export default [
    {
        plugins: { jsdoc },
        rules: {
            "jsdoc/no-undefined-types": "error"
        }
    }
];
```

### `index.js`

```javascript
/**
 * @import { Linter } from "eslint"
 */

/**
 * @type {Linter.Config}
 */
const config = {};
```

## To reproduce

1. `npm install`
2. `npx eslint index.js`

   ```
    /home/regseb/testcase/index.js
      6:1  error  The type 'Linter' is undefined  jsdoc/no-undefined-types

    ✖ 1 problem (1 error, 0 warnings)
   ```

## Environment

- Node version: 20.14.0
- ESLint version: 9.5.0
- `eslint-plugin-jsdoc` version: 48.2.13

--- a/docs/rules/no-undefined-types.md
+++ b/docs/rules/no-undefined-types.md
@@ -307,6 +307,36 @@ const a = new Todo();
  */
 // Settings: {"jsdoc":{"structuredTags":{"namepathOrURLReferencer":{"name":"namepath-or-url-referencing"}}}}
 // Message: The type 'SomeType' is undefined.
+
+/**
+ * @import BadImportIgnoredByThisRule
+ */
+/**
+ * @import LinterDef, { Sth as Something, Another as Another2 } from "eslint"
+ */
+/**
+ * @import { Linter } from "eslint"
+ */
+/**
+ * @import LinterDefault from "eslint"
+ */
+/**
+ * @import {Linter as Lintee} from "eslint"
+ */
+/**
+ * @import * as Linters from "eslint"
+ */
+
+/**
+ * @type {BadImportIgnoredByThisRule}
+ */
+/**
+ * @type {Sth}
+ */
+/**
+ * @type {Another}
+ */
+// Message: The type 'BadImportIgnoredByThisRule' is undefined.
 ````
 
 
@@ -778,7 +808,44 @@ function quux(foo) {
 quux(0);
 
 /**
+ * @import BadImportIgnoredByThisRule
+ */
+/**
+ * @import LinterDef, { Sth as Something, Another as Another2 } from "eslint"
+ */
+/**
  * @import { Linter } from "eslint"
+ */
+/**
+ * @import LinterDefault from "eslint"
+ */
+/**
+ * @import {Linter as Lintee} from "eslint"
+ */
+/**
+ * @import * as Linters from "eslint"
+ */
+
+/**
+ * @type {LinterDef}
+ */
+/**
+ * @type {Something}
+ */
+/**
+ * @type {Another2}
+ */
+/**
+ * @type {Linter}
+ */
+/**
+ * @type {LinterDefault}
+ */
+/**
+ * @type {Lintee}
+ */
+/**
+ * @type {Linters}
  */
 ````
 

--- a/package.json
+++ b/package.json
@@ -11,6 +11,7 @@
     "debug": "^4.3.4",
     "escape-string-regexp": "^4.0.0",
     "esquery": "^1.5.0",
+    "parse-imports": "^2.0.0",
     "semver": "^7.6.2",
     "spdx-expression-parse": "^4.0.0"
   },

--- a/src/rules/noUndefinedTypes.js
+++ b/src/rules/noUndefinedTypes.js
@@ -1,12 +1,13 @@
-import iterateJsdoc, {
-  parseComment,
-} from '../iterateJsdoc.js';
+import { parseImportsSync } from 'parse-imports';
 import {
   getJSDocComment,
   parse as parseType,
   traverse,
   tryParse as tryParseType,
 } from '@es-joy/jsdoccomment';
+import iterateJsdoc, {
+  parseComment,
+} from '../iterateJsdoc.js';
 
 const extraTypes = [
   'null', 'undefined', 'void', 'string', 'boolean', 'object',
@@ -109,13 +110,15 @@ export default iterateJsdoc(({
       .filter(Boolean));
   }
 
-  const typedefDeclarations = sourceCode.getAllComments()
+  const comments = sourceCode.getAllComments()
     .filter((comment) => {
       return (/^\*\s/u).test(comment.value);
     })
     .map((commentNode) => {
       return parseComment(commentNode, '');
-    })
+    });
+
+  const typedefDeclarations = comments
     .flatMap((doc) => {
       return doc.tags.filter(({
         tag,
@@ -126,6 +129,51 @@ export default iterateJsdoc(({
     .map((tag) => {
       return tag.name;
     });
+
+
+  const importTags = /** @type {string[]} */ (comments.flatMap((doc) => {
+    return doc.tags.filter(({
+      tag,
+    }) => {
+      return tag === 'import';
+    });
+  }).flatMap((tag) => {
+    const {
+      type, name, description
+    } = tag;
+    const typePart = type ? `{${type}} `: '';
+    const imprt = 'import ' + (description
+      ? `${typePart}${name} ${description}`
+      : `${typePart}${name}`);
+
+    let imports;
+    try {
+      // Should technically await non-sync, but ESLint doesn't support sync rules;
+      //  thankfully, the Wasm load time is safely fast
+      imports = parseImportsSync(imprt);
+    } catch (err) {
+      return null;
+    }
+
+    return [...imports].flatMap(({importClause}) => {
+      /* c8 ignore next */
+      const {default: dflt, named, namespace} = importClause || {};
+      const types = [];
+      if (dflt) {
+        types.push(dflt);
+      }
+      if (namespace) {
+        types.push(namespace);
+      }
+      if (named) {
+        for (const {binding} of named) {
+          types.push(binding);
+        }
+      }
+
+      return types;
+    });
+  }).filter(Boolean));
 
   const ancestorNodes = [];
 
@@ -193,6 +241,7 @@ export default iterateJsdoc(({
     )
     .concat(extraTypes)
     .concat(typedefDeclarations)
+    .concat(importTags)
     .concat(definedTypes)
     .concat(/** @type {string[]} */ (definedPreferredTypes))
     .concat(

--- a/test/rules/assertions/noUndefinedTypes.js
+++ b/test/rules/assertions/noUndefinedTypes.js
@@ -558,6 +558,52 @@ export default {
         },
       },
     },
+    {
+      code: `
+        /**
+         * @import BadImportIgnoredByThisRule
+         */
+        /**
+         * @import LinterDef, { Sth as Something, Another as Another2 } from "eslint"
+         */
+        /**
+         * @import { Linter } from "eslint"
+         */
+        /**
+         * @import LinterDefault from "eslint"
+         */
+        /**
+         * @import {Linter as Lintee} from "eslint"
+         */
+        /**
+         * @import * as Linters from "eslint"
+         */
+
+        /**
+         * @type {BadImportIgnoredByThisRule}
+         */
+        /**
+         * @type {Sth}
+         */
+        /**
+         * @type {Another}
+         */
+      `,
+      errors: [
+        {
+          line: 22,
+          message: 'The type \'BadImportIgnoredByThisRule\' is undefined.',
+        },
+        {
+          line: 25,
+          message: 'The type \'Sth\' is undefined.',
+        },
+        {
+          line: 28,
+          message: 'The type \'Another\' is undefined.',
+        },
+      ],
+    },
   ],
   valid: [
     {
@@ -1439,7 +1485,44 @@ export default {
     {
       code: `
         /**
+         * @import BadImportIgnoredByThisRule
+         */
+        /**
+         * @import LinterDef, { Sth as Something, Another as Another2 } from "eslint"
+         */
+        /**
          * @import { Linter } from "eslint"
+         */
+        /**
+         * @import LinterDefault from "eslint"
+         */
+        /**
+         * @import {Linter as Lintee} from "eslint"
+         */
+        /**
+         * @import * as Linters from "eslint"
+         */
+
+        /**
+         * @type {LinterDef}
+         */
+        /**
+         * @type {Something}
+         */
+        /**
+         * @type {Another2}
+         */
+        /**
+         * @type {Linter}
+         */
+        /**
+         * @type {LinterDefault}
+         */
+        /**
+         * @type {Lintee}
+         */
+        /**
+         * @type {Linters}
          */
       `,
     },

